### PR TITLE
Multiselect table

### PR DESCRIPTION
**Pull request by dr15**
- Adds multiselect functionality to `CoursesPage`.
- Shows dialogs when clicking on actions to mock flow (very partial).
- Improves publishing dropdown styling.
- Refactors `IWMDropdownOption` and updates anything that uses it.
- Hides unused filters

### CODE REVIEW

File: src/components/Screen/CoursesScreen/index.tsx [hunk 1]
```diff
@@ -86,25 +113,43 @@ export default function CoursesScreen(): ReactElement {
...
+              icon={<Icon type={IconType.Delete} />}
+              disabled={!hasSelected}
+              onClick={() => {
+                setShowDeleteCourse(true);
+              }}
+            />
+            <ExportButton
+              onClick={() => {
+                setShowExport(true);
+              }}
```
Comment (dvira) [inline src/components/Screen/CoursesScreen/index.tsx:151]: On the dialogs I saw you use the callbacks as one-liners when possible and here you didn't.
I vote for one-liners whenever possible every time! ;-)

File: src/components/common/WMDropdown/index.tsx [hunk 2]
```diff
@@ -42,7 +42,7 @@ export default function WMDropdown({
           ])}
           key={option.id}
         >
-          {option.text}
+          {option.label ?? option.value}
```
Comment (dvira) [inline src/components/common/WMDropdown/index.tsx:45]: Good thinking!

File: src/components/common/buttons/ExportButton/index.tsx [hunk 3]
```diff
@@ -1,17 +1,22 @@
...
 
 import classes from './style.module.scss';
 
-export default function ExportButton({ className }: { className?: string }): ReactElement {
+export default function ExportButton({
+  className,
+  onClick,
+}: {
+  className?: string;
+  onClick?: () => void; // todo: make this required
```
Comment (dvira) [inline src/components/common/buttons/ExportButton/index.tsx:14]: Meant to do this eventually, thanks :)

File: src/components/Screen/CoursesScreen/index.tsx (hunk 1 above)
Comment (nunibaranes) [reply src/components/Screen/CoursesScreen/index.tsx:151]: WE already discuss on this convention `onClick={() => setShowExport(true)}`?

File: src/components/Screen/CoursesScreen/index.tsx [hunk 4]
```diff
@@ -4,17 +4,21 @@ import { DownOutlined, PlusOutlined } from '@ant-design/icons';
...
+import SearchFilter from '../../common/filters/SearchFilter';
+import WMButton, { ButtonVariantEnum } from '../../common/WMButton';
+import WMCard from '../../common/WMCard';
+import WMDropdown, { IWMDropdownOption } from '../../common/WMDropdown';
+import WMTable from '../../common/WMTable';
+import WMTag from '../../common/WMTag';
+// dialogs
+import DeleteCourseDialog from '../../common/dialogs/DeleteCourseDialog';
+import DialogPublishToEnvironment from '../../common/dialogs/PublishToEnvironmentDialog';
+import ExportToCSVDialog from '../../common/dialogs/ExportToCSVDialog';
```
[the remark this answers is not in the document]
Comment (dr15) [reply src/components/Screen/CoursesScreen/index.tsx:21]: I agree, I was saving it fir a future refactor

File: src/components/Screen/CoursesScreen/index.tsx [hunk 5]
```diff
@@ -71,13 +94,17 @@ export default function CoursesScreen(): ReactElement {
...
+    if (selected.value === 'Published') setShowPublish(true);
+    else message.info(`Production status was changed to ${selected.value}`);
+  };
+
+  const onMultiSelectChange = (selectedRowKeys: any) => {
+    setSelectedRowKeys(selectedRowKeys);
+    setSelectedRows(table.data.filter((row) => selectedRowKeys.includes(row.key)));
   };
 
+  const hasSelected = selectedRowKeys.length > 0;
```
[the remark this answers is not in the document]
Comment (dr15) [reply src/components/Screen/CoursesScreen/index.tsx:107]: Changed to `const hasSelected = !!selectedRowKeys.length;`, fixed line spacing.

File: src/components/Screen/CoursesScreen/index.tsx (hunk 1 above)
Comment (dr15) [reply src/components/Screen/CoursesScreen/index.tsx:151]: Fixed

**Pull request merged**